### PR TITLE
Get TF:XLA working in docker

### PR DESCRIPTION
**Pull request by mariecwhite**
XLA:GPU and XLA:CPU baselines are being generated manually on a VM instance with similar hardware to the runners. This PR adds CUDA and CuDNN packages to the SHARK dockerfile in order for Tensorflow to run with XLA in the Docker container, removing the need to run benchmarks manually to get XLA baselines.

### CODE REVIEW

File: build_tools/docker/dockerfiles/shark.Dockerfile [hunk 1]
```diff
@@ -63,13 +81,5 @@ RUN wget -q \
...
 
 ENV VULKAN_SDK="/opt/vulkan-sdk/${VULKAN_SDK_VERSION}/x86_64"
-
 ENV PATH="${VULKAN_SDK}/bin:$PATH"
-
-# Symlink the Vulkan loader to a system library directory. This is needed to
-# allow Vulkan applications to find the Vulkan loader. It also avoids using
-# LD_LIBRARY_PATH, which is not supported well by Docker.
-RUN ln -s "${VULKAN_SDK}/lib/libvulkan.so" /usr/lib/x86_64-linux-gnu/ \
-  && ln -s "${VULKAN_SDK}/lib/libvulkan.so.1" /usr/lib/x86_64-linux-gnu/
```
[the remark this answers is not in the document]
Comment (mariecwhite) [reply build_tools/docker/dockerfiles/shark.Dockerfile:73]: The path already exists so it has probably been introduced by CUDA.

**Comment by mariecwhite** [comment]
> 

This can live outside of the IREE repo if we don't intend to use it for integration testing. The rationale for adding it here was that it may be valuable for running as a presubmit, postsubmit or before a release. SHARK does catch regressions regularly and we should be catching them before they make it into the codebase. We should also be steering towards a better developer/user experience.

**Comment by mariecwhite** [comment]
> > 
> 
> This can live outside of the IREE repo if we don't intend to use it for integration testing. The rationale for adding it here was that it may be valuable for running as a presubmit, postsubmit or before a release. SHARK does catch regressions regularly and we should be catching them before they make it into the codebase. We should also be steering towards a better developer/user experience.

Just had a discussion with @pzread and moving the benchmarking code to another repo is an option that has been considered before. All of the docker files, benchmarking code and external framework dependencies can live there. The IREE repo can then call into this repo during presubmit, etc. This will have to wait until Q1-Q2 though.

**Comment by pzread** [comment]
Link the related issue #10244

**Review by ScottTodd — APPROVED** [review]
Thanks. The timeline makes sense to me, I'll just keep voicing some discontent each time we change the core repo to support building TF/XLA/some other project :P

LGTM once the ` pull_request:` is removed

**Pull request merged**